addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py

```diff
@@ -369,6 +374,18 @@ def _invoice_constraints_cen_en16931_ubl(self, invoice, vals):
                 # /!\ exception: possible to have any number of ecotaxes (fixed tax) with a regular percentage tax
                 constraints.update({'cen_en16931_tax_line': _("Each invoice line shall have one and only one tax.")})
 
+        for role in ('supplier', 'customer'):
+            constraints[f'cen_en16931_{role}_country'] = self._check_required_fields(vals[role], 'country_id')
+            scheme_vals = vals['vals'][f'accounting_{role}_party_vals']['party_vals']['party_tax_scheme_vals'][-1:]
+            if not (scheme_vals and scheme_vals[0]['company_id'] and scheme_vals[0]['company_id'][:2].isalpha()) \
+                and (scheme_vals and scheme_vals[0]['tax_scheme_id'] == 'VAT') \
+                and self._name in ('account.edi.xml.ubl_bis3', 'account.edi.xml.ubl_nl', 'account.edi.xml.ubl_de'):
```

Comment:
```suggestion
if (
    not (scheme_vals and scheme_vals[0]['company_id'] and scheme_vals[0]['company_id'][:2].isalpha())
    and (scheme_vals and scheme_vals[0]['tax_scheme_id'] == 'VAT')
    and self._name in ('account.edi.xml.ubl_bis3', 'account.edi.xml.ubl_nl', 'account.edi.xml.ubl_de')
):
```
sorry but I can't read that part without vomiting :vomiting_face: